### PR TITLE
Implement new savebar design

### PR DESCRIPTION
**Pull request by dominik-zeglen**
I want to merge this change because it changes how savebar looks and behaves.
If form is unmodified, savebar should be docked at the bottom of the page.
If modified, then savebar should stick to the bottom of the window. After save, it should be docked again.

**PR intended to be tested with API branch:** master

### Pull Request Checklist

1. [x] All visible strings are translated with proper context.
1. [x] All data-formatting is locale-aware (dates, numbers, and so on).
1. [x] Translated strings are extracted.
1. [x] Number of API calls is optimized.
1. [x] The changes are tested.
1. [x] Data-test are added for new elements.
1. [x] Type definitions are up to date.
1. [ ] Changes are mentioned in the changelog.

### Test environment config

API_URI=https://master.staging.saleor.rocks/graphql/

### CODE REVIEW

**Comment by patrys** [comment]
Task linked: [SALEOR-1347 Implement design of the save bar](https://app.clickup.com/t/2549495/SALEOR-1347)

**Comment by mlamperski** [comment]
ISSUE: Save bar hides immediately after clicking Save button, after a second bar shows again for a moment. Issue occurs when trying to save changes on Metadata and Private Metadata fields, in e.g. Product, Product Type pages.
[Screen Recording 2020-10-30 at 17.10.08.zip](https://github.com/mirumee/saleor-dashboard/files/5466318/Screen.Recording.2020-10-30.at.17.10.08.zip)

System: macOs/Chrome
Env: Deployed test environment

**Comment by mlamperski** [comment]
Retested on deployed environment. Issue does not occur.

**Review by mlamperski — APPROVED** [review]

File: src/apps/components/CustomAppDetailsPage/CustomAppDetailsPage.tsx [hunk 1]
```diff
@@ -42,7 +42,7 @@ export interface CustomAppDetailsPageProps {
   onTokenDelete: (id: string) => void;
   onTokenClose: () => void;
   onTokenCreate: () => void;
-  onSubmit: (data: CustomAppDetailsPageFormData) => void;
+  onSubmit: (data: CustomAppDetailsPageFormData) => Promise<any>;
```
Comment (orzechdev) [inline src/apps/components/CustomAppDetailsPage/CustomAppDetailsPage.tsx:45]: `any` type in promise?

**Review by orzechdev — APPROVED** [review]

**Pull request merged**